service/controller/v25/resource/tccp/resource.go

```diff
@@ -104,75 +104,109 @@ func (r *Resource) Name() string {
-	return awstags.NewCloudFormation(tags)
-}
-
-func (r *Resource) newTemplateBody(ctx context.Context, customObject v1alpha1.AWSConfig, stackState StackState) (string, error) {
+func (r *Resource) searchMasterInstanceID(ctx context.Context, cr v1alpha1.AWSConfig) (string, error) {
 	cc, err := controllercontext.FromContext(ctx)
 	if err != nil {
 		return "", microerror.Mask(err)
 	}
 
-	cfg := adapter.Config{
-		APIWhitelist:                    r.apiWhiteList,
-		ControlPlaneAccountID:           cc.Status.ControlPlane.AWSAccountID,
-		ControlPlaneNATGatewayAddresses: cc.Status.ControlPlane.NATGateway.Addresses,
-		ControlPlanePeerRoleARN:         cc.Status.ControlPlane.PeerRole.ARN,
-		ControlPlaneVPCCidr:             cc.Status.ControlPlane.VPC.CIDR,
-		CustomObject:                    customObject,
-		EncrypterBackend:                r.encrypterBackend,
-		InstallationName:                r.installationName,
-		PublicRouteTables:               r.publicRouteTables,
-		Route53Enabled:                  r.route53Enabled,
-		StackState: adapter.StackState{
-			Name: stackState.Name,
-
-			DockerVolumeResourceName:   stackState.DockerVolumeResourceName,
-			MasterImageID:              stackState.MasterImageID,
-			MasterInstanceResourceName: stackState.MasterInstanceResourceName,
-			MasterInstanceType:         stackState.MasterInstanceType,
-			MasterCloudConfigVersion:   stackState.MasterCloudConfigVersion,
-			MasterInstanceMonitoring:   stackState.MasterInstanceMonitoring,
-
-			WorkerCloudConfigVersion: stackState.WorkerCloudConfigVersion,
-			WorkerDesired:            cc.Status.TenantCluster.TCCP.ASG.DesiredCapacity,
-			WorkerDockerVolumeSizeGB: stackState.WorkerDockerVolumeSizeGB,
-			WorkerImageID:            stackState.WorkerImageID,
-			WorkerInstanceMonitoring: stackState.WorkerInstanceMonitoring,
-			WorkerInstanceType:       stackState.WorkerInstanceType,
-			WorkerMax:                cc.Status.TenantCluster.TCCP.ASG.MaxSize,
-			WorkerMin:                cc.Status.TenantCluster.TCCP.ASG.MinSize,
-
-			VersionBundleVersion: stackState.VersionBundleVersion,
-		},
-		TenantClusterAccountID: cc.Status.TenantCluster.AWSAccountID,
-		TenantClusterKMSKeyARN: cc.Status.TenantCluster.KMS.KeyARN,
+	var instanceID string
+	{
+		i := &ec2.DescribeInstancesInput{
+			Filters: []*ec2.Filter{
+				{
+					Name: aws.String("tag:Name"),
+					Values: []*string{
+						aws.String(key.MasterInstanceName(cr)),
+					},
+				},
+				{
+					Name: aws.String("tag:giantswarm.io/cluster"),
+					Values: []*string{
+						aws.String(key.ClusterID(cr)),
+					},
+				},
+			},
+		}
+
+		o, err := cc.Client.TenantCluster.AWS.EC2.DescribeInstances(i)
+		if err != nil {
+			return "", microerror.Mask(err)
+		}
+
+		if len(o.Reservations) == 0 {
+			return "", microerror.Maskf(notExistsError, "master instance")
+		}
+		if len(o.Reservations) != 1 {
+			return "", microerror.Maskf(executionFailedError, "expected one master instance, got %d", len(o.Reservations))
```

Comment:
My thinking was that the stack tells you which line the error was about. I find it good that way. Let me know when you want me to change it or feel free to provide a PR.

Reply:
I'm fine with this in the sense that most of our logs require intervention to code anyway. This is still part of thing I've said. You always need to look to code in order to make anything useful out of a log line and in my opinion that's bad.

Reply:
Since you need to check code anyway always I cannot really comprehend that argument. I do not want you to just swallow the practise though and it might be good if you could bring it up with the team. Maybe in the sync, in case this is important. Don't want you to make more unhappy than necessary. 🤗